test/e2e/scenario/recorder.scenario.ts

```diff
@@ -654,17 +694,305 @@ describe('recorder', () => {
+      .withRum({ enableExperimentalFeatures: ['visualviewport'] })
+      .withRumInit(initRumAndStartRecording)
+      .withSetup(bundleSetup)
+      .withBody(html`${VIEWPORT_META_TAGS}`)
+      .run(async ({ events }) => {
+        if (isGestureUnsupported()) {
+          return // No Fallback test possible
+        }
+
+        await resetViewport()
+
+        const initialVisualViewportDimension = await getVisualViewport()
+        await pinchZoom(170)
+        await sleep(210)
+        const nextVisualViewportDimension = await getVisualViewport()
+
+        await flushEvents()
+        const segment = getLastSegment(events)
+        const visualViewportRecords = findAllVisualViewport(segment)
+        const lastVisualViewportRecord = visualViewportRecords.slice(-1)[0]
+
+        // SDK returns Visual Viewport object
+        expectToBeNearby(lastVisualViewportRecord.data.scale, nextVisualViewportDimension.scale)
+        expectToBeNearby(lastVisualViewportRecord.data.width, nextVisualViewportDimension.width)
+        expectToBeNearby(lastVisualViewportRecord.data.height, nextVisualViewportDimension.height)
+        expectToBeNearby(lastVisualViewportRecord.data.offsetLeft, nextVisualViewportDimension.offsetLeft)
+        expectToBeNearby(lastVisualViewportRecord.data.offsetTop, nextVisualViewportDimension.offsetTop)
+        expectToBeNearby(lastVisualViewportRecord.data.pageLeft, nextVisualViewportDimension.pageLeft)
+        expectToBeNearby(lastVisualViewportRecord.data.pageTop, nextVisualViewportDimension.pageTop)
+
+        // With correct transformation
+        const finalScaleAmount = nextVisualViewportDimension.scale
+        expectToBeNearby(lastVisualViewportRecord.data.scale, finalScaleAmount)
+        expectToBeNearby(lastVisualViewportRecord.data.width, initialVisualViewportDimension.width / finalScaleAmount)
+        expectToBeNearby(lastVisualViewportRecord.data.height, initialVisualViewportDimension.height / finalScaleAmount)
+
+        expect(lastVisualViewportRecord.data.offsetLeft).toBeGreaterThan(0)
+        expect(lastVisualViewportRecord.data.offsetTop).toBeGreaterThan(0)
+        expect(lastVisualViewportRecord.data.pageLeft).toBeGreaterThan(0)
+        expect(lastVisualViewportRecord.data.pageTop).toBeGreaterThan(0)
+      })
+  })
 })
 
-function getFirstSegment(events: EventRegistry) {
+export function getFirstSegment(events: EventRegistry) {
   return events.sessionReplay[0].segment.data
 }
 
-function getLastSegment(events: EventRegistry) {
+export function getLastSegment(events: EventRegistry) {
   return events.sessionReplay[events.sessionReplay.length - 1].segment.data
 }
 
-function initRumAndStartRecording(initConfiguration: RumInitConfiguration) {
+export function initRumAndStartRecording(initConfiguration: RumInitConfiguration) {
   window.DD_RUM!.init(initConfiguration)
   window.DD_RUM!.startSessionReplayRecording()
 }
+
+export async function pinchZoom(xChange = 50, durationMS = 400) {
```

Comment:
why xChange=50?
it seems that most of the time you are using it with 150

you never use the durationMS parameter so it could be another constant inside the function